File: javascript/core/src/engine/verbs/dedupe.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { DedupeArgs } from '../../types.js'
import { makeStepFunction, makeStepNode, wrapColumnStep } from '../factories.js'

/**
 * Executes an arquero dedupe operation.
 */
const doDedupe = wrapColumnStep<DedupeArgs>((input, { columns }) =>
	columns ? input.dedupe(columns) : input.dedupe(),
)

export const dedupe = makeStepFunction(doDedupe)
export const dedupeNode = makeStepNode(doDedupe)

File: python/datashaper/datashaper/verbs/dedupe.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Dedupe verb implementation."""

from typing import Any

import pandas as pd

from .decorators import OutputMode, inputs, verb, wrap_verb_result


@verb(
    name="dedupe",
    immutable_input=True,
    adapters=[
        inputs(default_input_argname="table"),
        wrap_verb_result(mode=OutputMode.Table),
    ],
)
def dedupe(
    table: pd.DataFrame, columns: list[str] | None = None, **_kwargs: Any
) -> pd.DataFrame:
    """Dedupe verb implementation."""
    return table.drop_duplicates(columns)
